src/sentry/tasks/auto_enable_codecov.py

```diff
@@ -83,8 +91,12 @@ def enable_for_org(dry_run: bool = False) -> None:
 def disable_codecov_access(
     organization: Organization, integration_enabled: bool, task_enabled: bool
 ) -> None:
-    organization.flags.codecov_access = False
-    organization.save()
+
+    updated_flag = F("flags") - Organization.flags.codecov_access
+    update_organization_with_outbox_message(
+        org_id=organization.id, update_data={"flags": updated_flag}
+    )
```

Comment:
These changes honestly don't feel great. I'm not a fan of forcing a more complicated way of setting/unsetting flags on an org 😬

Reply:
Should we be making the new interface take a dict of flags & their states so that calling code doesn't have to build F() expressions?

```python
update_organization_with_outbox_message(
   org_id=organization.id,
   update_data={"flags": {"codecov_access": False}}
)
```

Flag names can be validated/read from the `organization.flags` attribute. We could likely use mypy to typecheck dict keys as well.